yjit/src/asm/x86_64/mod.rs

```diff
@@ -972,16 +972,7 @@ pub fn mov(cb: &mut CodeBlock, dst: X86Opnd, src: X86Opnd) {
                 write_opcode(cb, 0xB8, reg);
                 cb.write_int(uimm.value, 32);
             } else {
-                if reg.num_bits == 16 {
-                    cb.write_byte(0x66);
-                }
-
-                if dst.rex_needed() || reg.num_bits == 64 {
-                    write_rex(cb, reg.num_bits == 64, 0, 0, reg.reg_no);
-                }
-
-                write_opcode(cb, if reg.num_bits == 8 { 0xb0 } else { 0xb8 }, reg);
-                cb.write_int(uimm.value, reg.num_bits.into());
+                movabs(cb, dst, uimm.value);
```

Comment:
I don't think you should have changed the code for `mov` here? Just introducing `movabs` which you can call explicitly should be sufficient?

Reply:
OK, I resurrected the original `mov` implementation https://github.com/ruby/ruby/pull/6733/commits/25fdeeffb3f36e9cab238da16d6663acf2b19695 and just made this code specialized to the new usage https://github.com/ruby/ruby/pull/6733/commits/333be1d50923ac4db26460661e7aceb16485129d.